extensions/rich_text_components/Image/directives/oppia-noninteractive-image.directive.ts

```diff
@@ -98,12 +98,22 @@ angular.module('oppia').directive('oppiaNoninteractiveImage', [
             // don't have loading indicator or try again for showing images in
             // this case. So we directly assign the url to the imageUrl.
             try {
+              // For IMAGE_SAVE_DESTINATION_LOCAL_STORAGE mode we first try to
+              // fetch images through the storage and if it doesn't exist we try
+              // fetching it through server.
+              // This is required for the translation suggestion as there can be
+              // target entity's images in the translatable content which needs
+              // to be fetched from the server.
+              ctrl.imageUrl === null;
               if (
                 ContextService.getImageSaveDestination() ===
-                IMAGE_SAVE_DESTINATION_LOCAL_STORAGE) {
+                IMAGE_SAVE_DESTINATION_LOCAL_STORAGE && (
+                  ImageLocalStorageService.isInStorage(ctrl.filepath))) {
                 ctrl.imageUrl = ImageLocalStorageService.getObjectUrlForImage(
                   ctrl.filepath);
-              } else {
+              }
+
+              if (ctrl.imageUrl === null) {
```

Comment:
This could just be an "else", right?

Or is it possible for getObjectUrlForImage() above to return null?

Reply:
Earlier I had this structure later on I added isFileInStorage method but I had not refactored this code. I've fixed it now!